solutions/USACO dec16-cow-checklist.mdx

```diff
@@ -0,0 +1,88 @@
+---
+id: usaco-670
+source: USACO Gold 2016 December
+title: Cow Checklist
+author: Maggie Liu
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_checklist_gold_dec16.html)
+
+<LanguageSection>
+
+<CPPSection>
+
+**Time Complexity:** $\mathcal{O}(H \cdot G)$
+
+```cpp
+#include <iostream>
+#include <cstdio>
+#include <cmath>
+using namespace std;
+
+int energy(pair<int, int> a, pair<int, int> b)
+{
+	return pow(a.first - b.first, 2) + pow(a.second - b.second, 2);
+}
+
+int main() 
+{
+	freopen("checklist.in", "r", stdin);
+	freopen("checklist.out", "w", stdout);
+	int h, g;
+	cin >> h >> g;
+	pair<int, int> hcows[h], gcows[g];
```

Comment:
As noted [here](https://usaco.guide/bronze/intro-ds?lang=cpp#dynamic-arrays) variable-length arrays aren't part of the C++ standard, so I would prefer using a vector for this purpose.